Patches/Android Tiers/Ammo/MastiffShotgunShell.xml

```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+
+	<Operation Class="PatchOperationFindMod">
+		<mods>
+			<li>[1.0] Android tiers</li>
+		</mods>
+		<match Class="PatchOperationSequence">
+			<operations>
+				<li Class="PatchOperationAdd">
+					<xpath>Defs</xpath>
+					<value>
+
+						<!-- There is only one type of Mastiff shell for CE:FT, no subcategory for variants required -->
+
+						<!-- ==================== AmmoSet ========================== -->
+
+						<CombatExtended.AmmoSetDef>
+							<defName>AmmoSet_MastiffShotgunShell</defName>
+							<label>Mastiff shell</label>
+							<ammoTypes>
+								<Ammo_MastiffShotgunShell>Bullet_MastiffShotgunShell</Ammo_MastiffShotgunShell>
+							</ammoTypes>
+						</CombatExtended.AmmoSetDef>
+
+						<!-- ==================== Ammo ========================== -->
+
+						<ThingDef Class="CombatExtended.AmmoDef" Name="MastiffShotgunShellBase" ParentName="AmmoBase" Abstract="True">
+							<description>Very large caliber buckshot shell designed specifically for the mech-operated Mastiff Shotgun.</description>
+							<statBases>
+								<Mass>2.128</Mass>
+								<Bulk>7.22</Bulk>
+							</statBases>
+							<tradeTags>
+								<li>CE_AutoEnableTrade</li>
+								<li>CE_AutoEnableCrafting_TableMachining</li>
+							</tradeTags>
+							<thingCategories>
+								<li>AmmoAdvanced</li>
+							</thingCategories>
+							<stackLimit>25</stackLimit>
+						</ThingDef>
+
+						<ThingDef Class="CombatExtended.AmmoDef" ParentName="MastiffShotgunShellBase">
+							<defName>Ammo_MastiffShotgunShell</defName>
+							<label>Mastiff shell (Buck)</label>
+							<graphicData>
+								<texPath>ThirdParty/Android Tiers/MastiffShell</texPath>
+								<graphicClass>Graphic_StackCount</graphicClass>
+							</graphicData>
+							<statBases>
+								<MarketValue>9.52</MarketValue>
+							</statBases>
+							<ammoClass>BuckShot</ammoClass>
+						</ThingDef>
+
+						<!-- ================== Projectiles ================== -->
+
+						<ThingDef Class="CombatExtended.AmmoDef" Name="BaseMastiffShotgunShell" ParentName="BaseBullet" Abstract="true">
```

Comment:
Check your base here; you're using the ammo item base, when you should probably be using BaseBullet or something similar.

Reply:
Um, I'm not entirely sure what you mean...

```
<ThingDef Class="CombatExtended.AmmoDef" Name="BaseMastiffShotgunShell" ParentName="BaseBullet" Abstract="true">
```

If I understand correctly, the `ParentName` right now is already BaseBullet, which is the same as most (if not all) of the projectiles defined core CE library:

```
<!-- 12 Gauge shotgun shells -->
<ThingDef Class="CombatExtended.AmmoDef" Name="Base12GaugeBullet" ParentName="BaseBullet" Abstract="true">

<!-- 90mm Cannon shells -->
<ThingDef Class="CombatExtended.AmmoDef" Name="Base90mmCannonShell" ParentName="BaseBullet" Abstract="true">
```

Reply:
Ah, my mistake. At a glance I was mixing up `BaseMastiffShotgunShell` and `MastiffShotgunShellBase`.

Reply:
Since `BaseMastiffShotgunShell` fits the established naming format for the projectile base of the current caliber, I'm going to assume this particular line of XML is a non-issue.

Reply:
Fixed in 564bb6f, e2561a9 and 1b5cde9 

See [this comment](https://github.com/CombatExtendedRWMod/CombatExtended/pull/1058#issuecomment-595764201) for details.